experimental/AlgebraicStatistics/src/MultigradedImplicitization.jl

```diff
@@ -0,0 +1,326 @@
+    exps = [exponent_vector(f,i) for i in 1:length(f)]
+    eqs = matrix(ZZ, vcat([exps[1] - exps[i] for i in 2:length(exps)]))
+    homogeneity_eqs = vcat(homogeneity_eqs, eqs)
+  end
+  
+  return rref(transpose(nullspace(homogeneity_eqs)[2]))[2]
+end
+
+@doc raw"""
+    max_grading(phi::MPolyAnyMap)
+
+Computes the maximal rank grading on the kernel of $\phi: \mathbb{K}[x_1, \ldots, x_n] \to \mathbb{K}[t_1, \ldots, t_m]$ which can be induced by grading the codomain of `phi`. 
+The first $m$ columns of the resulting matrix correspond to the degrees of the codomain while the last $n$ columns are the resulting degrees on the domain. 
+
+## Examples
+
+```jldoctest
+julia> R, x = polynomial_ring(QQ, :x => (1:2, 1:2))
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[x[1, 1] x[1, 2]; x[2, 1] x[2, 2]])
+
+julia> S, s, t = polynomial_ring(QQ, :s => 1:2, :t => 1:2)
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[s[1], s[2]], QQMPolyRingElem[t[1], t[2]])
+
+julia> phi = hom(R, S, [s[1]*t[1], s[1]*t[2], s[2]*t[1], s[2]*t[2]])
+Ring homomorphism
+  from multivariate polynomial ring in 4 variables over QQ
+  to multivariate polynomial ring in 4 variables over QQ
+defined by
+  x[1, 1] -> s[1]*t[1]
+  x[2, 1] -> s[1]*t[2]
+  x[1, 2] -> s[2]*t[1]
+  x[2, 2] -> s[2]*t[2]
+
+julia> max_grading(phi)
+[1   0   0   0   1   1   0   0]
+[0   1   0   0   0   0   1   1]
+[0   0   1   0   1   0   1   0]
+[0   0   0   1   0   1   0   1]
+```
+"""
+function max_grading(phi::MPolyAnyMap)
+  codom = codomain(phi)
+  dom = domain(phi)
+  elim_ring, z = polynomial_ring(QQ, vcat(symbols(codom), symbols(dom)))
+  lift_codom = hom(codom, elim_ring, gens(elim_ring)[1:ngens(codom)])
+  lift_dom = hom(dom, elim_ring, z[ngens(codom) + 1:ngens(elim_ring)])
+
+  return homogeneity_space(ideal([lift_dom(x) - lift_codom(phi(x)) for x in gens(dom)]))
+end
+
+@doc raw"""
+    jacobian(phi::MPolyAnyMap)
+
+Computes the Jacobian of a polynomial map $\phi: \mathbb{K}[x_1, \ldots, x_n] \to \mathbb{K}[t_1, \ldots, t_m]$ such that $(i, j)$ entry is $\frac{d \phi(x_j)}{d t_i}$.
+
+## Examples
+
+```jldoctest
+julia> R, x = polynomial_ring(QQ, :x => (1:2, 1:2))
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[x[1, 1] x[1, 2]; x[2, 1] x[2, 2]])
+
+julia> S, s, t = polynomial_ring(QQ, :s => 1:2, :t => 1:2)
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[s[1], s[2]], QQMPolyRingElem[t[1], t[2]])
+
+julia> phi = hom(R, S, [s[1]*t[1], s[1]*t[2], s[2]*t[1], s[2]*t[2]])
+Ring homomorphism
+  from multivariate polynomial ring in 4 variables over QQ
+  to multivariate polynomial ring in 4 variables over QQ
+defined by
+  x[1, 1] -> s[1]*t[1]
+  x[2, 1] -> s[1]*t[2]
+  x[1, 2] -> s[2]*t[1]
+  x[2, 2] -> s[2]*t[2]
+
+julia> jacobian(phi)
+[t[1]   t[2]      0      0]
+[   0      0   t[1]   t[2]]
+[s[1]      0   s[2]      0]
+[   0   s[1]      0   s[2]]
+```
+"""
+function jacobian(phi::MPolyAnyMap)
```

Comment:
```suggestion
function jacobian(phi::MPolyAnyMap{<:MPolyRing, <:MPolyRing})
```
If we really want to restrict to honest polynomial maps, we probably want to do this?